src/pages/Workspace/components/Actions/index.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  useReactivateSubscriptionMutation,
+  useRenewCancelSubscriptionMutation,
+} from "@api";
+import { CustomModal } from "@elements";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import { SubscriptionInactiveStatuses, SubscriptionResponse } from "@types";
+import { useState, type FC } from "react";
+import { useTranslation } from "react-i18next";
+import days from "dayjs";
+
+type Props = {
+  subscription: SubscriptionResponse;
+};
+
+export const Actions: FC<Props> = ({ subscription }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [type, setType] = useState<"cancel" | "renew" | "reactivate">("cancel");
+  const { t } = useTranslation("", { keyPrefix: "pages.workspace" });
+  const [reactivateSubscription] = useReactivateSubscriptionMutation();
+  const [renewCancelSubscription] = useRenewCancelSubscriptionMutation();
+
+  const actionsMap = {
```

Comment:
You dont  need this  map  .. also type, setType .   For  buttons  renew -cancel  -  use  one  method  renewCancelSubscription,  and  label  for  button  would  be  rendered  depending  on  subscription  cancelAtPerioEnd  flag

Reply:
For  reactivating subscription You dint  need  actionsMap - just  check  subscription  status